forwarder/internal/logs/models.go

```diff
@@ -21,6 +22,31 @@ import (
 	customtime "github.com/DataDog/azure-log-forwarding-orchestration/forwarder/internal/time"
 )
 
+var supportedTimeLayouts []string
+
+func init() {
+	// Add the supported time layouts for parsing Azure logs
+	supportedTimeLayouts = []string{
+		time.RFC3339,          // ISO 8601 format,
+		"01/02/2006 15:04:05", // USA short timestamp format
+		"1/2/2006 3:04:05 PM", // USA short timestamp format with AM/PM
+	}
+}
+
+func parseTime(timeString string) (parsedTime time.Time, timeParsingErrors error) {
+	timeString = strings.TrimSpace(timeString) // Trim leading and trailing whitespace
+	for _, layout := range supportedTimeLayouts {
+		var currErr error
+		parsedTime, currErr = time.Parse(layout, timeString)
+		if currErr == nil {
+			timeParsingErrors = nil
+			break // Successfully parsed the time
```

Comment:
Less verbose

```suggestion
			return parsedTime, nil
```

And at the end of the function, instead of the naked return, an explicit one. You can even remove the named return variables.

```go
return time.Time{}, timeParsingErrors
```